extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -1684,6 +1689,22 @@ public enum ResponseMode {
          */
         public boolean failOnMissingStateParam = false;
 
+        /**
+         * Fail with the HTTP 401 error if the ID token signature can not be verified during the re-authentication only due to
+         * an unresolved token key identifier (`kid`).
+         * <p>
+         * This property might need to be disabled when multiple tab authentications are allowed, with one of the tabs keeping
+         * an expired ID token with its `kid`
+         * unresolved due to the verification key set refreshed due to another tab initiating an authorization code flow. In
+         * such cases, instead of failing with the HTTP 401 error,
+         * redirecting the user to re-authenticate with the HTTP 302 status may provide better user experience.
+         * <p>
+         * Note that the HTTP 401 error is always returned if the ID token signature can not be verified due to an unresolved
+         * kid during an initial ID token verification
+         * following the authorization code flow completion, before a session cookie is created.
+         */
+        public boolean failOnUnresolvedKid = true;
```

Comment:
IMHO private would be correct visibility, it is a new field and if someone needs to set it, let them use builder

Reply:
I was watching this PR before, but I didn't mention it, sorry.